meteor-core/src/main/java/dev/pixelib/meteor/core/proxy/PendingInvocation.java

```diff
@@ -52,17 +52,15 @@ public void complete(Object response) throws IllegalStateException {
         }
 
         boolean isVoidOrNullable = invocationDescriptor.getReturnType().equals(Void.TYPE) || !invocationDescriptor.getReturnType().isPrimitive();
-
+        boolean isInstance = invocationDescriptor.getReturnType().isInstance(response);
+        boolean isPrimitive = invocationDescriptor.getReturnType().isPrimitive();
+        boolean isBoxed = ArgumentMapper.ensureBoxedClass(invocationDescriptor.getReturnType()).isInstance(response);
```

Comment:
I agree that this does make the code more readable, but it also causes unnecessary map lookups, where it would otherwise never reach this branch.